onnxruntime/core/framework/session_options.h

```diff
@@ -81,5 +81,8 @@ struct SessionOptions {
 
   // Deterministic compute is likely not as performant. This option is default to false.
   bool use_deterministic_compute = false;
+
+  // Stores the configurations for this session
+  std::unordered_map<int, std::string> session_configurations;
```

Comment:
Could use the name of the enum as the key instead of int to be clear.

Reply:
I did this in my previous commit, which compiles fine on my machine, but failed linux CI build, since I did not define the hash function for the key, for simplicity, I switched to use int dirrectly

---
In reply to: [466694949](https://github.com/microsoft/onnxruntime/pull/4718#discussion_r466694949) [](ancestors = 466694949)

Reply:
Why not define the hash function? Even something like this (from a SO answer) would be fine in this case (I don't think we need for a 'real' hash)

struct EnumClassHash
{
    template <typename T>
    std::size_t operator()(T t) const
    {
        return static_cast<std::size_t>(t);
    }
};

---
In reply to: [466752787](https://github.com/microsoft/onnxruntime/pull/4718#discussion_r466752787) [](ancestors = 466752787,466694949)